x/wasm/types/types.go

```diff
@@ -83,6 +90,60 @@ func (c *ContractInfo) ValidateBasic() error {
 	if err := validateLabel(c.Label); err != nil {
 		return sdkerrors.Wrap(err, "label")
 	}
+	if c.Extension == nil {
+		return nil
+	}
+	e, ok := c.Extension.GetCachedValue().(validatable)
+	if !ok {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "extension: %T", c.Extension)
+	}
+	if err := e.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(err, "extension")
+	}
+	return nil
+}
+
+// SetExtension set new extension data
+func (c *ContractInfo) SetExtension(ext ContractInfoExtension) error {
+	if e, ok := ext.(validatable); ok {
+		if err := e.ValidateBasic(); err != nil {
+			return err
+		}
+	}
+	if ext == nil {
```

Comment:
Shouldn't this be the first line of this function? (quick handling of nil, as with ValidateBasic above)